documentation/release_notes.markdown

```diff
@@ -6,6 +6,68 @@ canonical: "/puppetdb/latest/release_notes.html"
 
 [configure_postgres]: ./configure.html#using-postgresql
 [pg_trgm]: http://www.postgresql.org/docs/current/static/pgtrgm.html
+[upgrading]: ./api/query/v4/preparing-for-3.0.html
+
+2.3.4
+-----
+
+PuppetDB 2.3.4 is a backwards-compatible bugfix release that addresses an issue
+that would prevent fact storage for nodes under certain circumstances.
+
+### Upgrading
+
+* For the best-possible performance and scaling capacity, we recommend
+  PostgreSQL version 9.4 or newer with the [`pg_trgm`][pg_trgm]
+  extension enabled, as explained [here][configure_postgres], and we
+  have officially deprecated versions earlier than 9.2.  HSQLDB is
+  only recommended for local development because it has a number of
+  scaling and operational issues.
+
+* Make sure that all of your PuppetDB instances are shut down, and
+  only upgrade one at a time.
+
+* Make sure to upgrade your puppetdb-terminus package (on the host
+  where your Puppet master lives), and restart your master service.
+
+### Contributors
+
+Andrew Roetker, Deepak Giridharagopal, Ken Barber, Melissa Stone, Rob Browning,
+Wyatt Alt
+
+### Changes
+
+#### Bug Fixes and Maintenance
+
+* We have fixed a bug where if any of the facts for a given node exchange values
+within a single puppet run, and none of values are referred to by another node,
+the facts update for the node would fail. ([PDB-1448](https://tickets.puppetlabs.com/browse/PDB-1448))
+
```

Comment:
@wkalt not sure it's worth worrying about, but thought about rearranging this to "...where the facts update for a node would fail if any of the nodes facts change within a single puppet run, and none of those facts..." i.e. move the final clause toward the front somehow.